src/worker/vm/extensions/fetch.ts

```diff
@@ -0,0 +1,80 @@
+import dns from 'dns'
+import ipRangeCheck from 'ip-range-check'
+import net from 'net'
+import nodeFetch, { HeaderInit, RequestInit, Response } from 'node-fetch'
+import url from 'url'
+
+import { IllegalOperationError } from '../../plugins/run'
+
+export interface URLLike {
+    href: string
+}
+
+// ['93.184.216.34', '172.31.0.0/16']
+const RESTRICTED_IP_RANGES =
+    process.env.NODE_ENV === 'test'
+        ? ['127.0.0.1/8', '116.203.255.68']
+        : (process.env.RESTRICTED_IP_RANGES || '').split(',')
+
+const isIpInRestrictedRange = (ip: string): boolean => {
+    return ipRangeCheck(ip, RESTRICTED_IP_RANGES)
+}
+
+const validateHostOrUrl = async (hostOrUrl: any) => {
```

Comment:
Could we have _unit_ tests for this function? That way we could cover cases really cleanly (in addition the existing tests checking that this is indeed raised in VMs)